waiter/integration/waiter/request_timeout_test.clj

```diff
@@ -223,7 +223,7 @@
                                                  (quot 1000000))] ; truncated nanos->millis
               (is (<= startup-delay-ms instance-acquired-delay-ms)
                   (str "Healthy instance was found in just " instance-acquired-delay-ms " ms (too short)")))
-            (when (using-marathon? waiter-url)
+            (when (can-query-for-grace-period? waiter-url)
               (is (= (t/in-seconds grace-period) (service-id->grace-period waiter-url service-id)))))
```

Comment:
`service-id->grace-period` should be fixed so it can retrieve it for both marathon and kubernetes. The `check-pod-namespace` has an example of retricing data from the pod state.

Reply:
separate PR

Reply:
updated